nixos/modules/hardware/dell-command-configure.nix

```diff
@@ -0,0 +1,66 @@
+{ config, lib, pkgs, ... }:
+
+with lib;
+
+let
+  cfg = config.hardware.dellCommandConfigure;
+  
+  configFile = pkgs.writeTextFile {
+    name = "omreg.cfg";
+    text = ''
+      hapi.omilcore.version=9.3.0
+      hapi.configtool=${pkgs.dell-command-configure}/bin/dchcfg
+      hapi.installpath=${pkgs.dell-command-configure}
+      hapi.logpath=${cfg.logpath}
+      hapi.vardatapath=${cfg.vardatapath}
+
+      openmanage.openipmi.kernel.2.4.x.ver_min_major=35
+      openmanage.openipmi.kernel.2.4.x.ver_min_minor=13
+      openmanage.openipmi.kernel.2.6.x.ver_min_major=33
+      openmanage.openipmi.kernel.2.6.x.ver_min_minor=13
+      openmanage.openipmi.kernel.ver_min_major=2
+      openmanage.openipmi.kernel.ver_min_minor=6
+      openmanage.openipmi.kernel.ver_min_patch=15
+      openmanage.openipmi.rhel3.ver_min_major=35
+      openmanage.openipmi.rhel3.ver_min_minor=13
+      openmanage.openipmi.rhel4.ver_min_major=33
+      openmanage.openipmi.rhel4.ver_min_minor=13
+    '';
+  };
+in
+
+{
+  options = {
+    hardware.dellCommandConfigure = {
+      enable = mkEnableOption "Dell Command Configure";
+
+      logpath = mkOption {
+        type = types.path;
+        default = "/var/log/openmanage";
+        description = "Log path.";
+      };
+
+      vardatapath = mkOption {
+        type = types.path;
+        default = "/var/lib/openmanage";
+        description = "Data directory path.";
+      };
+    };
+  };
+
+  config = mkIf config.hardware.dellCommandConfigure.enable {
+    environment.systemPackages = [ pkgs.dell-command-configure ];
+
+    system.activationScripts.dellCommandConfigure =
+      ''
```

Comment:
hum I believe activationScript should be avoided as it can easily break things . Also the script creates states as the folders won't be removed afterwards
I am no systemd expert but greping nixpkgs, it can create folders in a way that's more fit.
```
        RuntimeDirectoryPreserve = "yes";
        LogsDirectory = subDirs [ "qemu" ];
        RuntimeDirectory = subDirs [ "nix-emulators" "nix-helpers" "nix-ovmf" ];
        StateDirectory = subDirs [ "dnsmasq" ];
```

Reply:
Managing directories with a oneshot systemd unit felt weird, but so did the solution I went for. I guess I would end up here with something like?
```
ExecStartPre = "mkdir -p /opt/dell/srvadmin/etc";
ExecStart = "ln -fs ${configFile} /opt/dell/srvadmin/etc/omreg.cfg";
LogsDirectory = "openmanage";
StateDirectory = "openmanage";
```
And remove the configuration options? Though I am unclear on how the symlink and directory would be cleaned up in this case.